### PR TITLE
feat(server): add support for core plugin

### PR DESCRIPTION
**Pull request by pyshx**
Marketplace Renovations

## What I've done

1. I will add the core flag to the plugin data.
```diff
type Plugin implements Node {
...
+  core: Boolean
}
```

2. I will add the core flag to the parameters when creating a plugin.
```diff
input CreatePluginInput {
...
+  core: Boolean
}
```

3. Implemented the base code for E2E testing

## What I haven't done
1. E2E tests do not support real file operations
2. Web code needs to be revised

## How I tested
Implemented E2E test and confirmed operation
https://github.com/reearth/reearth-marketplace/pull/91/files#diff-e5d3b21865847fbcf74214a5708df678b29172199a3fcdd38eec22ebe7117f03R15

## Which point I want you to review particularly
This is implemented manually without using a GraphQL generator.
It is guaranteed to work through E2E testing, but you need to check for any issues.

## Release Notes

- **New Features**
  - Introduced a new `Core` field in the `Plugin` type, enhancing plugin management capabilities.
  - Added pagination support for `plugins` methods in `User` and Organization types.
  - New GraphQL scalar type `Lang` added for language codes.
  - Enhanced end-to-end testing setup with new methods for server initialization and seeding test data.
  - New Makefile targets for common development tasks, including `lint`, `test`, and `build`.

- **Bug Fixes**
  - Corrected import paths for `mongodoc` in multiple files to ensure proper functionality.

- **Tests**
  - Added tests for core plugin functionality, including retrieval and creation scenarios.
  - Improved test coverage for end-to-end scenarios, including ping and GraphQL plugin retrieval.

- **Chores**
  - Updated dependencies for improved compatibility and functionality.

### CODE REVIEW

File: server/e2e/common.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+)
+
+type GraphQLRequest struct {
+	OperationName string         `json:"operationName"`
+	Query         string         `json:"query"`
+	Variables     map[string]any `json:"variables"`
+}
+
+func init() {
+	mongotest.Env = "REEARTH_DB"
```
Comment (pyshx) [inline server/e2e/common.go:27]: REEARTH_MARKETPLACE_DB

File: server/internal/infrastrcture/fs/file.go
Comment (pyshx) [inline server/internal/infrastrcture/fs/file.go:1]: infrastructure

*Spelling mistake*

File: server/e2e/common.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	if err := mongorepo.InitRepos(ctx, repos, db.Client(), "reearth_marketplace", mongox.IsTransactionAvailable(conf.DB)); err != nil {
+		log.Fatalf("Failed to init mongo: %+v\n", err)
+	}
+
+	// File
+	// fileRepo, err := gcs.NewFile(conf.GCS.Bucket, conf.GCS.AssetsBucket, conf.GCS.AssetsBaseURL)
+	fileRepo, err := gcs.NewFile("marketplace.test.reearth.dev", "assets.marketplace.test.reearth.dev", "")
+	if err != nil {
+		log.Fatalf("file: init error: %v", err)
+	}
```
[the remark this answers is not in the document]
Comment (pyshx) [reply server/e2e/common.go]: @hexaforce Please do this.

File: server/internal/infrastrcture/fs/file.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		pluginBucketName: bucketName,
+		assetsBucketName: assetsBucketName,
+		assetsBaseURL:    assetsBaseURL,
+	}, nil
+}
+
+func (f *fileRepo) UploadPlugin(ctx context.Context, vid id.VersionID, content []byte) error {
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (pyshx) [reply server/internal/infrastrcture/fs/file.go:35]: @hexaforce Can you add a TODO comment.

File: server/internal/infrastrcture/fs/file.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+func (f *fileRepo) DownloadPlugin(ctx context.Context, vid id.VersionID) (io.ReadCloser, error) {
+
+	return nil, nil
+}
```
[the remark this answers is not in the document]
Comment (pyshx) [reply server/internal/infrastrcture/fs/file.go:40]: Same comment.

File: server/internal/infrastrcture/fs/file.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+func (f *fileRepo) UploadImage(ctx context.Context, image io.ReadSeeker) (string, error) {
+
+	return "", nil
+}
```
[the remark this answers is not in the document]
Comment (pyshx) [reply server/internal/infrastrcture/fs/file.go:45]: Same comment

File: server/e2e/common.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (hexaforce) [reply server/e2e/common.go]: yes

File: server/e2e/common.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+	"context"
+	"log"
+	"net"
+	"net/http"
+	"testing"
+
+	"github.com/gavv/httpexpect/v2"
+	"github.com/reearth/reearth-marketplace/server/internal/app"
+	"github.com/reearth/reearth-marketplace/server/internal/infrastrcture/gcs"
+	mongorepo "github.com/reearth/reearth-marketplace/server/internal/infrastrcture/mongo"
```
[the remark this answers is not in the document]
Comment (hexaforce) [reply server/e2e/common.go]: yes

File: server/e2e/common.go (hunk 1 above)
Comment (hexaforce) [reply server/e2e/common.go:27]: yes

File: server/internal/infrastrcture/fs/file.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (hexaforce) [reply server/internal/infrastrcture/fs/file.go:45]: yes

File: server/internal/infrastrcture/fs/file.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (hexaforce) [reply server/internal/infrastrcture/fs/file.go:40]: yes

**Review by hexaforce — APPROVED** [review]

**Pull request merged**